src/Microsoft.AspNetCore.Mvc.Core/Internal/FileStreamResultExecutor.cs

```diff
@@ -18,18 +23,72 @@ public FileStreamResultExecutor(ILoggerFactory loggerFactory)
 
         public Task ExecuteAsync(ActionContext context, FileStreamResult result)
         {
-            SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            var fileLength = 0L;
```

Comment:
If you can't get the length then you don't set content-length. Write the body normally and the server will set transfer-encoding: chunked.

You may not be able to calculate/verify ranges without the length,